src/Cli/dotnet/CommonArguments.cs

```diff
@@ -0,0 +1,51 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+
+using System.CommandLine;
+using System.CommandLine.Parsing;
+using Microsoft.DotNet.Cli.Utils;
+using NuGet.Packaging.Core;
+using NuGet.Versioning;
+
+namespace Microsoft.DotNet.Cli
+{
+    internal class CommonArguments
+    {
+        #region PackageIdentityArgument
+        public static CliArgument<PackageIdentity> PackageIdentityArgument(bool requireArgument = true) =>
+            new("packageId")
+            {
+                HelpName = "PACKAGE_ID",
+                Description = CliStrings.PackageIdentityArgumentDescription,
+                CustomParser = ParsePackageIdentity,
+                Arity = requireArgument ? ArgumentArity.ExactlyOne : ArgumentArity.ZeroOrOne,
+            };
+
+        private static PackageIdentity ParsePackageIdentity(ArgumentResult argumentResult)
+        {
+            if (argumentResult.Tokens.Count == 0)
+            {
+                return null;
+            }
+            var token = argumentResult.Tokens[0].Value;
+            var versionSeparatorIndex = token.IndexOf('@');
+            if (versionSeparatorIndex == -1)
+            {
+                return new(token, null);
+            }
+            var packageId = token.Substring(0, versionSeparatorIndex);
+            var versionString = token.Substring(versionSeparatorIndex + 1);
+            if (string.IsNullOrEmpty(packageId) || string.IsNullOrEmpty(versionString))
```

Comment:
I _think_ we have other things that treat package@ the same as package, so we should probably do that here as well

Reply:
![image](https://github.com/user-attachments/assets/d10c610f-3ef4-4d5d-9b28-6b99c23b25dc)
Refactored to support this :)

Reply:
That feels weird to me. @Forgind where do you think we support this?

Reply:
https://github.com/dotnet/sdk/blob/d91d8dac4f9e33fd29b8a47bf80957becbb57ce8/test/dotnet-new.Tests/DotnetNewInstallTests.cs#L53

`dotnet new install package::` was supported in the past, so when I changed :: to @, I included that `dotnet new install package@` should work.

Looking to see how long `dotnet new install package::` has been supported, it came into the SDK via #26521, which sounded like it just brought it here from templates, so it's been in the product for at least 4 years, probably longer.